packages/skill-api/next-auth.d.ts

```diff
@@ -1,12 +1,20 @@
 import NextAuth, {DefaultSession} from 'next-auth'
 
+type Resolve<T> = T extends Function ? T : {[K in keyof T]: T[K]}
+
+type User = Resolve<
+  {
+    role?: string
+    purchases?: any[]
+  } & DefaultSession['user']
+>
```

Comment:
This expands the NextAuth `Session` / `User` type to allow `user` to include things like `role` and `purchases`.